spring-cloud-aws-secrets-manager/src/main/java/io/awspring/cloud/secretsmanager/SecretsManagerPropertySource.java

```diff
@@ -94,7 +101,8 @@ private void readSecretValue(GetSecretValueRequest secretValueRequest) {
 			String[] parts = secretValueResponse.name().split("/");
 			String secretName = parts[parts.length - 1];
 			LOG.debug("Populating property retrieved from AWS Secrets Manager: " + secretName);
-			properties.put(secretName, secretValueResponse.secretString());
+			properties.put(secretName, secretValueResponse.secretString() != null ? secretValueResponse.secretString()
+					: secretValueResponse.secretBinary().asUtf8String());
```

Comment:
if i get it right we assume here that the secret is a string. what if it is really a binary secret like a certificate that can't be represented as a string? I guess this won't work then, right?

Reply:
Hmmm interesting will try that out tomorrow.